internal/db/schema/migrations/oss/postgres/59/01_ldap.up.sql

```diff
@@ -0,0 +1,354 @@
+'auth_ldap_method has at least one auth_ldap_url';
+
+create constraint trigger auth_ldap_url_children_per_parent_tg
+  after insert or update or delete on auth_ldap_url deferrable initially deferred 
+  for each row execute procedure auth_ldap_url_parent_children();
+
+create or replace function auth_ldap_method_children() returns trigger as $$
+declare
+  n integer;
+begin
+  if tg_op = 'INSERT' then 
+    select into n count(*) from auth_ldap_url where ldap_method_id = new.public_id;
+     if n < 1 then 
+      raise exception 'During % of auth_ldap_method public_id=% must have at least one url, not %',tg_op,new.public_id,n;
+    end if;
+    -- No need for an UPDATE or DELETE check, as regular referential integrity constraints
+    -- and the trigger on `child' will do the job.
+
+    return null;
+  end if;
+end;
+$$ language plpgsql;
+comment on function auth_ldap_method_children() is 
+'function used on auth_ldap_method after insert initially deferred to ensure each '
+'auth_ldap_method has at least one auth_ldap_url';
+
+create constraint trigger auth_ldap_method_children_tg
+  after insert on auth_ldap_method deferrable initially deferred 
+  for each row execute procedure auth_ldap_method_children();
+
+-- auth_ldap_user_entry_search entries specify the required parameters to find a
+-- user entry before attempting to authenticate the user.
+--
+-- Updates will be implemented as a delete + insert with the auth_ldap_method
+-- being used as the root aggregate for auth_ldap_user_entry_search updates.
+create table auth_ldap_user_entry_search (
+  create_time wt_timestamp,
+  auth_ldap_method_id wt_public_id not null
+    constraint auth_ldap_method_fkey
+      references auth_ldap_method(public_id)
+      on delete cascade
+      on update cascade,
+  user_dn text
+    constraint user_dn_too_short 
+      check (length(trim(user_dn)) > 0)
+    constraint user_dn_too_long
+      check (length(trim(user_dn)) < 1025),
+  user_attr text
+    constraint user_attr_too_short
+      check (length(trim(user_attr)) > 0)
+    constraint user_attr_too_long
+      check (length(trim(user_attr)) < 1025),
+  user_filter text
+    constraint user_filter_too_short
+      check (length(trim(user_filter)) < 0)
+    constraint user_filter_too_long
+      check (length(trim(user_filter)) < 2049)
+);
+comment on table auth_ldap_user_entry_search is 
+'auth_ldap_user_entry_search entries specify the required parameters to find '
+'a user entry before attempting to authenticate the user';
+
+-- enforce a zero or one-to-one relationship between auth_ldap_method and
+-- auth_ldap_user_entry_search 
+create unique index auth_ldap_user_entry_search_zero_or_one_idx
+on auth_ldap_user_entry_search(auth_ldap_method_id) 
+  where auth_ldap_method_id is not null;
+
+-- TODO -trigger to prevent all auth_ldap_user_entry_search fields from being null
+
+create table auth_ldap_group_entry_search (
```

Comment:
Can you add a `create comment` for this table?

Reply:
happy to.

Reply:
happy too!

Reply:
Can you add it? :)

Reply:
Ha!  sure.